testsuite/model/src/test/java/org/keycloak/testsuite/model/infinispan/SplitBrainTest.java

```diff
@@ -0,0 +1,100 @@
+package org.keycloak.testsuite.model.infinispan;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.keycloak.connections.infinispan.InfinispanConnectionProvider.WORK_CACHE_NAME;
+
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.junit.Test;
+import org.keycloak.connections.infinispan.InfinispanConnectionProvider;
+import org.keycloak.models.Constants;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.RealmModel;
+import org.keycloak.models.RealmProvider;
+import org.keycloak.models.UserProvider;
+import org.keycloak.models.UserSessionProvider;
+import org.keycloak.models.cache.CacheRealmProvider;
+import org.keycloak.models.session.UserSessionPersisterProvider;
+import org.keycloak.testsuite.model.KeycloakModelTest;
+import org.keycloak.testsuite.model.RequireProvider;
+
+@RequireProvider(CacheRealmProvider.class)
+@RequireProvider(InfinispanConnectionProvider.class)
+@RequireProvider(RealmProvider.class)
+@RequireProvider(UserSessionPersisterProvider.class)
+@RequireProvider(UserSessionProvider.class)
+@RequireProvider(UserProvider.class)
```

Comment:
Maybe all of these are overkill? I wasn't really sure what the consequences were of adding/omitting Providers here.

Reply:
The tests are run for different configuration mutations as listed in the different profiles which lead to differently configured setups which are based on KeycloakModelParameters. Nowadays, all profiles are based on JPA of some kind, which provides for example the RealmProvider. Still, not all profiles include Infinispan, so the first two are definitely relevant.